src/app/core/entity-components/entity-select/entity-select/entity-select.component.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  Component,
+  Input,
+  Output,
+  EventEmitter,
+  ViewChild,
+  ElementRef,
+  TemplateRef,
+  OnChanges,
+  SimpleChanges,
+} from "@angular/core";
+import { ENTER, COMMA } from "@angular/cdk/keycodes";
+import { Entity, EntityConstructor } from "../../../entity/entity";
+import { EntityMapperService } from "../../../entity/entity-mapper.service";
+import { BehaviorSubject, Observable } from "rxjs";
+import { FormControl } from "@angular/forms";
+import { filter, map, skipWhile } from "rxjs/operators";
+import { MatChipInputEvent } from "@angular/material/chips";
+import { LoggingService } from "../../../logging/logging.service";
+
+export type accessorFn<E extends Entity> = (E) => string;
+
+@Component({
+  selector: "app-entity-select",
+  templateUrl: "./entity-select.component.html",
+  styleUrls: ["./entity-select.component.scss"],
+})
+export class EntitySelectComponent<E extends Entity> implements OnChanges {
+  /**
+   * The type of entity to load. This is required and will cause all
+   * entities of the given type to be available in the selection
+   * and auto-complete
+   * @param type The type of the entity
+   */
+  @Input() set entityType(type: EntityConstructor<E>) {
+    this.entityMapperService
+      .loadType<E>(type)
+      .then((entities) => {
+        this.allEntities = entities;
+        this.loading.next(false);
+        this.formControl.setValue(null);
+      })
+      .catch((error) => {
+        this.loggingService.warn(error);
+        this.loading.next(false);
+      });
+  }
+  /**
+   * The (initial) selection. Can be used in combination with {@link selectionChange}
+   * to enable two-way binding to either an array of entities or an array of strings
+   * corresponding to the id's of the entities.
+   * <br>
+   * <b>Important:</b> If nothing is passed initially, then the type of this will be id's (in other
+   * words, id's will be emitted whenever something changes).
+   * If this behavior is not desired, the {@link selectionInputType} has to be set to "entity" explicitly
+   * These id's have to be legal id's for the given {@link entityType}
+   * @param sel The initial selection
+   */
+  @Input() set selection(sel: (string | E)[]) {
+    if (this.selectionInputType === "id") {
```

Comment:
It could be if the selection is always non-empty. If it is empty, there is no way to find out if it's a string or an entity array (unless I'm missing something). This leads to problems when there is an empty initial array (for example when creating a new note) and then a new entity is added. Should this new entity be a string or an entity?